From 40101e4da939fcea2eebe3e4800d0de4e551ca26 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Wed, 1 Jul 2020 08:44:07 +0800
Subject: [PATCH] policy/modules/services/rpcbind: allow rpcbind_t to create
 directory with label rpcbind_runtime_t

* Allow rpcbind_t to create directory with label rpcbind_runtime_t
* Set context for nfsserver and nfscommon

Fixes:
avc:  denied  { create } for  pid=136 comm="rpcbind" name="rpcbind"
scontext=system_u:system_r:rpcbind_t:s0-s15:c0.c1023
tcontext=system_u:object_r:var_run_t:s0 tclass=dir permissive=0

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/rpc.fc     | 2 ++
 policy/modules/services/rpcbind.te | 5 +++--
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/policy/modules/services/rpc.fc b/policy/modules/services/rpc.fc
index 88d2acaf0..d9c0a4aa7 100644
--- a/policy/modules/services/rpc.fc
+++ b/policy/modules/services/rpc.fc
@@ -1,7 +1,9 @@
 /etc/exports	--	gen_context(system_u:object_r:exports_t,s0)
 
 /etc/rc\.d/init\.d/nfs	--	gen_context(system_u:object_r:nfsd_initrc_exec_t,s0)
+/etc/rc\.d/init\.d/nfsserver	--	gen_context(system_u:object_r:nfsd_initrc_exec_t,s0)
 /etc/rc\.d/init\.d/nfslock	--	gen_context(system_u:object_r:rpcd_initrc_exec_t,s0)
+/etc/rc\.d/init\.d/nfscommon	--	gen_context(system_u:object_r:rpcd_initrc_exec_t,s0)
 /etc/rc\.d/init\.d/rpcidmapd	--	gen_context(system_u:object_r:rpcd_initrc_exec_t,s0)
 
 /usr/bin/rpc\..*	--	gen_context(system_u:object_r:rpcd_exec_t,s0)
diff --git a/policy/modules/services/rpcbind.te b/policy/modules/services/rpcbind.te
index 370c9bce6..8972980fa 100644
--- a/policy/modules/services/rpcbind.te
+++ b/policy/modules/services/rpcbind.te
@@ -25,16 +25,17 @@ files_type(rpcbind_var_lib_t)
 # Local policy
 #
 
-allow rpcbind_t self:capability { dac_override setgid setuid sys_tty_config };
+allow rpcbind_t self:capability { dac_override setgid setuid sys_tty_config chown };
 # net_admin is for SO_SNDBUFFORCE
 dontaudit rpcbind_t self:capability net_admin;
 allow rpcbind_t self:fifo_file rw_fifo_file_perms;
 allow rpcbind_t self:unix_stream_socket { accept listen };
 allow rpcbind_t self:tcp_socket { accept listen };
 
+manage_dirs_pattern(rpcbind_t, rpcbind_runtime_t, rpcbind_runtime_t)
 manage_files_pattern(rpcbind_t, rpcbind_runtime_t, rpcbind_runtime_t)
 manage_sock_files_pattern(rpcbind_t, rpcbind_runtime_t, rpcbind_runtime_t)
-files_runtime_filetrans(rpcbind_t, rpcbind_runtime_t, { file sock_file })
+files_runtime_filetrans(rpcbind_t, rpcbind_runtime_t, { file sock_file dir })
 
 manage_dirs_pattern(rpcbind_t, rpcbind_var_lib_t, rpcbind_var_lib_t)
 manage_files_pattern(rpcbind_t, rpcbind_var_lib_t, rpcbind_var_lib_t)
-- 
2.17.1

